### DISCUSSION TITLE
MACROSILICON HDMI CAPTURE DEVICE (534D:2109) Woes...

### DISCUSSION THREAD

**Discussion by EvoSems** [awawa-dev/HyperHDR#107] [General]
I know this isn't an HyperHDR issue. but I wanted to see if other people have got this problem and have a working solution..?

So I ordered a Capture card from Aliexpress and all was great! https://a.aliexpress.com/_mNaR9Pt

So here is the device in action on my test setup working with Pi4/ESP8266/HyperWLED/sk6812@AWA2000000Baud Perfectly 
https://youtu.be/2cbUlYUvygo 

the next day after I had it working perfectly it just didn't register on the Pi anymore. did a fresh install and tried everything. it just won't register in the devices. also tried on Mac with HyperHDR too. the Capture card initially shows as MACROSILICON HDMI CAPTURE DEVICE (534D:2109)

I'm assuming its either cooked itself or something? It Still turns on. and I get HDMI Passthrough. tried many cables too.

Anyone have any experience with the same or a solution even?

Many Thanks (p.s I have ordered an ezcap320 for a replacement)

RPi Stats (HyperHDR won't let me select anything other than Automatic for Device / Beta17)

```
pi@RaspberryPi4:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 004: ID 0451:16a8 Texas Instruments, Inc. 
Bus 001 Device 003: ID 1a86:7523 QinHeng Electronics HL-340 USB-Serial adapter
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
pi@RaspberryPi4:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 004: ID 0451:16a8 Texas Instruments, Inc. 
Bus 001 Device 006: ID 534d:2109  
Bus 001 Device 003: ID 1a86:7523 QinHeng Electronics HL-340 USB-Serial adapter
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
pi@RaspberryPi4:~ $ ls -l /dev/serial/by-id
total 0
lrwxrwxrwx 1 root root 13 Aug 26 23:37 usb-1a86_USB_Serial-if00-port0 -> ../../ttyUSB0
lrwxrwxrwx 1 root root 13 Aug 26 23:37 usb-Texas_Instruments_TI_CC2531_USB_CDC___0X00124B001CCC9657-if00 -> ../../ttyACM0
pi@RaspberryPi4:~ $ v4l2-ctl --list-devices
USB3. 0 capture: USB3. 0 captur (usb-0000:01:00.0-1.3):
	/dev/video1

Cannot open device /dev/video0, exiting.
pi@RaspberryPi4:~ $ dmesg|grep uvcvideo
[  481.595909] uvcvideo: Found UVC 1.00 device USB3. 0 capture (534d:2109)
[  481.601197] usbcore: registered new interface driver uvcvideo
pi@RaspberryPi4:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 004: ID 0451:16a8 Texas Instruments, Inc. 
Bus 001 Device 003: ID 1a86:7523 QinHeng Electronics HL-340 USB-Serial adapter
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
pi@RaspberryPi4:~ $
```

**Comment by awawa-dev** [comment]
Hi
Sorry to hear that. Yes, if the device is not detected by the Kernel then probably at least the grabber's part inside died. Did you try with all HDMI cables disconnected (to get the 'no signal' board or if it appears then in the devices at least) and without any other USB device connected to the Rpi? There is know issue with MACROSILICON grabbers that some manufactures remove internal radiators to reduce the cost. Of course don't know if that's the case but it seems it's a custom design with internal splitter...there is always a place for bad design in such cases also.  
You should be happy with Ezcap 320, just remember to avoid firmware upgrades if not necessary (latest firmware from may completely breaks my 320) and to try out custom LUT table for that grabber 😉 
Regards  
Awawa

**Comment by EvoSems** [comment]
Just wanted to say what great work you have done for everyone with HyperHDR!

I just checked as you instructed it still doesn't show up with the kernel. oh well. I will try my luck with a Aliexpress refund.... lol

Ah ok. I will not update at all then, thank you . do you know if cec works with the 320?

would you be kind to link me to out tables. I had not even thought about that, I'll get to researching some more.

kind regards

Jim

**Comment by awawa-dev** [comment]
Unfortunately CEC doesn't work for my ezcap 320...but my model may be from some kind of first, defective batch. You can find the LUT table in the latest stable v16 [link](https://github.com/awawa-dev/HyperHDR/releases/download/v16.0.0.1/EZCAP_320_lut_table_for_1080p_hdr_correction.tar.xz). I will provide it (will copy it in fact as it doesn't change :wink: ) when there will be official v17 stable release.
Best regards
Awawa

**Reply by EvoSems** [reply]
many thanks! = )

**Comment by lillebroeer** [comment]
hey i did the firmware upgrade :(
where can i get a working firmware for the ezcap 320 ?

**Reply by awawa-dev** [reply]
Write to their support and ask for older 320_1_3_8_4_1230 version , they reply fast. I think I still have a backup of that firmware but must search for it. If you won't be able to obtain the firmware from the support, let me know.

**Reply by mjoshd** [reply]
@awawa-dev 
I recently received an Ezcap320 from Ali Express and it captures some devices (computer & cable box) but refuses to capture others (blu-ray player & chromecast) so HyperHDR won't show anything for those last two. The strange part is that the video passthrough works for all devices. Do you think the 320_1_3_8_4_1230 firmware will solve that issue or the problem is something else? 

Previously I was using https://www.amazon.com/gp/product/B08CMXLN88 which worked for all 4 devices but only output mjpeg format and I wanted to use the Ezcap320 so I could select yuyv with a higher resolution.

**Reply by awawa-dev** [reply]
Please precise what are the symptoms: HyperHDR doesn't capture any video stream and it reports it in the logs or it captures the black board?  Windows or Linux?

Judging which devices refuse to work I suspect HDCP may come in play. But it should show HDCP error in the capture preview (some text on the black board). Or the blue-ray player & chromecast try to use DolbyVision (I don't have DV player so don't know how it reacts..usually grabbers refuse to capture the video stream or are providing pink disordered content. Mostly firmware downgrade is required for Raspberry Pi when Ezcap fails to initialize the Linux v4l2 capturing process at all, which is quite serious with no other workaround (probably bug in the firmware or lack of full support from v4l2 driver).

**Reply by mjoshd** [reply]
Thanks for the quick reply. 
I moved the Ezcap320 to a computer, instead of the Rpi, and then opened the camera app and I was able to see the output from the Ezcap stating that it was an HDCP issue. I will try to find a way to strip HDCP from the cable that goes to the Ezcap320.

**Reply by awawa-dev** [reply]
Probably your previous grabber and for example Ezcap 269 can strip HDCP, so I don't know why they change it with Ezcap 320. On the other hand these devices could support older HDCP version and Ezcap 320 supports one of the latest which could affect player capabilities (may restrict higher quality content only for the newest HDCP). Anyway most of Chinese HDMI splitters/matrixes can strip that protection without reporting it to the player.

**Reply by elpeterson** [reply]
I just picked up an Ezcap 326 (only one I could find) from AliExpress and I'm getting the HDCP Protection warning even when I just plug it into my Nvidia Shield (the only output I'll use it for). Do you have a specific splitter you'd recommend? After waiting 30 days for this to arrive only for it to not work I want to make sure I get the right splitter

**Comment by mjoshd** [comment]
As it turns out, the [splitter I already have](https://www.amazon.com/gp/product/B07VP37KMB) apparently has this ability but it was not working for me. 

I reached out to their support and they said the updated firmware on the [device webpage](https://www.easycoolav.com/products/4k60-hdmi-splitter-1x2-dolby-vision-hdr-scaler-edid-setting-sp12h2) (v2.0.6) has corrected the issue. 

I updated the firmware and it's now working as desired 😸 

Thanks for your help @awawa-dev !

**Comment by EvoSems** [comment]
Hi guys. new oled tv. so I made a new ambilight setup with the ezycap 320 (timber surround). Is there a preferred firmware version that works best? looking for the best hdr results. did the latest firmware update but fear I made a mistake. does anyone have a firmware library for this device?
[<a href="https://ibb.co/30k42Db"><img src="https://i.ibb.co/dQ7jFdv/432-C8093-7-D93-4804-BB36-75-D998-C9-DA39.jpg" alt="432-C8093-7-D93-4804-BB36-75-D998-C9-DA39" border="0"></a>](url) My New Timber Surround

**Comment by awawa-dev** [comment]
Nice TV frame :) Why do you want to update Ezcap 320? Is something broken? Updating this grabber is risky as newer software doesn't always mean better. Not to mention, the update manual was messed up causing only false/partial update, but it looks like they finally updated it. Had serious problem with 320_1.4.3_0519 firmware on Linux (the grabber refuses to capture on USB2.0, worked fine on USB3.0 on Windows) and I asked them for older 320_1_3_8_4_1230

**Reply by EvoSems** [reply]
Honestly. I just updated for the sake of it. It still works fine. I just feel the colours are not amazing. I see from the v18 lut ranking. 320 has a low score. 

P.s I have ordered a new splitter. With high hopes I can still use CEC with the Apple TV. I will report back my findings.

**Reply by EvoSems** [reply]
Thanks for the frame! I planned to use diffuser. But might use an acrylic pour to complete. The diffuser was too hard to use on the corner joints. 
<a href="https://ibb.co/9nGCrpJ"><img src="https://i.ibb.co/TcrFgwZ/095-DC027-7675-46-F7-B5-EB-3533-BB88-C7-A2.jpg" alt="095-DC027-7675-46-F7-B5-EB-3533-BB88-C7-A2" border="0"></a><br /><a target='_blank' href='https://imgbb.com/'>share pic online</a><br />

**Reply by awawa-dev** [reply]
You could try old LUT table for Ezcap 320: https://github.com/awawa-dev/HyperHDR/releases/download/v16.0.0.1/EZCAP_320_lut_table_for_1080p_hdr_correction.tar.xz
It was created manually using different approach.

**Comment by EvoSems** [comment]
https://www.amazon.com/dp/B09B7B7BM8

Confirmed working with CEC on hdmi 1 out. I use hdmi 2 out for grabber. No hacks. Just works.
